### DISCUSSION TITLE
Sysex for I2C display

### DISCUSSION THREAD

**Discussion by nicolasvair** [shanteacontrols/OpenDeck#220] [General]
Hi, 

First thank you for your fantastic work. I've been wanting to build a midi controller since a long time and your boards really helped jump in.

I am working with a opendeck M v2 and planned to display message from daw using sysex. Is it possible ? I thought it would be ok through sysex but now realize sysex might only be for configuration purpose... 

Isn't it any way to send information on that screen ? 

Thank you for your feedback.

**Comment by paradajz** [comment]
Do you want the display to show you the exact received SysEx message or just that it has received it?

**Reply by nicolasvair** [reply]
I’d like to show a message that I would set through sysex. 

For example :   
“scene : 2”

Ableton (through max for live) would send a sysex tailored to show this message on the display. 

Then I could have a lot of different information from Ableton. 

I just need a way to communicate with the display, I thought it was already implemented through sysex.

**Reply by paradajz** [reply]
Ah I see, no, sadly that is not possible.

**Comment by nicolasvair** [comment]
Thats a shame, the screen is quite useless without it… 

How complicated would it be to implement it ? 

Could you please point me to the code part that relate to this ? (Or the closest like the part that display midi out messages)

Thank you

**Reply by paradajz** [reply]
https://github.com/shanteacontrols/OpenDeck/blob/master/src/firmware/application/io/i2c/peripherals/display/Display.cpp#L448

**Reply by nicolasvair** [reply]
Hi paradajz,

I'm considering editing the code to have access to screen. How do you test the code modification prior to building it ?
Also regarding compilation, do I absolutely need ubuntu or can I use macos as it's also unix ? 

Finally, I guess I'll need another arduino to flash the board is that correct or is it possible to generate a sysex firmware ?

**Reply by paradajz** [reply]
> How do you test the code modification prior to building it ?

Huh? You mean test before flashing? You need to build the code first.

 > Also regarding compilation, do I absolutely need ubuntu or can I use macos as it's also unix ?

I don't use macOS nor do I support it for development anymore so you're on your own if you go that route.

> Finally, I guess I'll need another arduino to flash the board

Depends on which board you're talking about.

> or is it possible to generate a sysex firmware

SysEx firmware is automatically generated during the build process.

**Reply by nicolasvair** [reply]
> SysEx firmware is automatically generated during the build process.

wow that's the best news of today. So if I manage to build the code, it's only a matter of uploading sysex. Great. (I'm using the opendeck M Board)


> Huh? You mean test before flashing? You need to build the code first.

Isn't there any way to test the behavior on computer ? (I mean without the board)

Thanks

**Reply by paradajz** [reply]
> Isn't there any way to test the behavior on computer ? (I mean without the board)

There are no tests for the display currenly.

**Comment by nicolasvair** [comment]
On the other side, I found a workaround that could be ok in the meantime, which is to use midi values like C1 for chorus, A1 for verse 1 B1 for break etc
It works great (except a small bug that relate octave and velocity value), but my problem is that I'm also using leds, so there is a lot of midi information related to led control that makes this workaround not usable.

I'm wondering if it could be possible to sort what is shown on the OLED, for exemple using midi channel (without having to rebuild a firmware myself). Any idea ?

**Reply by eurorackworld** [reply]
I'm thinking about programming a 2nd micro controller (Arduino) and decode the midi messages. You can easily then map labels or whatever you want to do with those messages in that 2nd controller.

There is already an old script here which is a bit what I want to check how that would work; https://youthvulture.blogspot.com/2013/02/arduino-midi-message-decoder.html

**Comment by nicolasvair** [comment]
Hi, that’s really great ! 

In the meantime I found a workaround using TouchOSC on iPhone and using it to show the information I need. 
It is so easy to update that I think I’ll stick with it so far. 

But I might use your script later, after addressing more urgent issues. 

Thanks a lot for taking the time to answer !!
